### ISSUE TITLE
i consistently get spurious "module not found" errors

### ISSUE THREAD

**Issue by ryanking** [wata727/tflint#148]
`ERROR:` module `git@github.com:chanzuckerberg/shared-infra//terraform/modules/aws-env?ref=master` not found. Did you run `terraform get`?

From line:

https://github.com/wata727/tflint/blob/5865a74a904ebdbc26387ba671a005861142b234/schema/module.go#L45

My current setup works so it seems that this check is too naive?

**Comment by wata727** [comment]
Umm...  I could not reproduce it in my environment. (Terraform: v0.10.6, TFLint: v0.4.3)
For example, the following configuration works correctly.

```hcl
module "ecs_on_spotfleet" {
  source = "github.com/wata727/tf_aws_ecs_on_spotfleet"

  vpc         = "vpc-12345678"
  subnets     = ["subnet-12345678", "subnet-abcd1234"]
  spot_prices = ["0.700000", "0.008900"]
  key_name    = "demo-app"
}

output "endpoint" {
  value = "${module.ecs_on_spotfleet.endpoint}"
}
```

```
$ terrafrom init
Downloading modules...
Get: git::https://github.com/wata727/tf_aws_ecs_on_spotfleet.git
...
$ tflint
template.tf
    WARNING:2 Module source "github.com/wata727/tf_aws_ecs_on_spotfleet" is not pinned (terraform_module_pinned_source)

Result: 1 issues  (0 errors , 1 warnings , 0 notices)
```

Please tell me about following:

- Terraform version
- TFLint version
- `ls -l .terraform/modules` output

**Comment by ryanking** [comment]
terraform 0.10.7
tflint 0.4.3

It seems that this problem happens with git (not github) and local path references, but I haven't had time to narrow it down beyond that.

Here's some more info:

```
16:21 $ terraform get
✔ ~/code/idseq-infra/terraform/envs/alpha/cloud-env [ryanking/shrink_cluster L|⚑ 2] 
16:21 $ tflint
ERROR: module `git@github.com:chanzuckerberg/shared-infra//terraform/modules/aws-env?ref=master` not found. Did you run `terraform get`?
✘-2 ~/code/idseq-infra/terraform/envs/alpha/cloud-env [ryanking/shrink_cluster L|⚑ 2] 
16:21 $ ls -l .terraform/modules/
total 24
drwxr-xr-x  12 rking  staff  408 Oct  4 11:28 00eea973aaf2fc2361293b1eecb981a1
drwxr-xr-x  12 rking  staff  408 Oct  4 11:36 060210b1f32edc0bb1d92871274a1fae
lrwxr-xr-x   1 rking  staff  148 Oct  9 14:24 166b57b70fb2f29998d836c9e5ec40f4 -> /Users/rking/code/idseq-infra/terraform/envs/alpha/cloud-env/.terraform/modules/aaf523a4b3f807928cf15988c2e1e0f7/modules/consul-security-group-rules
drwxr-xr-x  11 rking  staff  374 Oct  9 14:24 196113a62150a0bde00e807e875a1dd8
lrwxr-xr-x   1 rking  staff  140 Oct  9 14:24 1ab73d0b46b58cdf6541e656fbcc1d99 -> /Users/rking/code/idseq-infra/terraform/envs/alpha/cloud-env/.terraform/modules/aaf523a4b3f807928cf15988c2e1e0f7/modules/consul-iam-policies
drwxr-xr-x  17 rking  staff  578 Oct  4 11:04 aaf523a4b3f807928cf15988c2e1e0f7
drwxr-xr-x  12 rking  staff  408 Oct  4 11:06 bfc10befc76ae7611526bcc87ba7c852
drwxr-xr-x  12 rking  staff  408 Oct  4 11:04 e942fc660873db760e9196792a07d33d
drwxr-xr-x  12 rking  staff  408 Oct  4 11:36 f25f135c0d54a2a9d48cb134ea1cf01a
-rw-r--r--   1 rking  staff  490 Oct  9 14:24 module-subdir.json
✔ ~/code/idseq-infra/terraform/envs/alpha/cloud-env [ryanking/shrink_cluster L|⚑ 2] 
16:21 $ 
```

**Comment by hposca** [comment]
@wata727 Copying/pasting your own example I could not get the same result:

```
hugo:lint/ $ terraform init
Downloading modules...
Get: git::https://github.com/wata727/tf_aws_ecs_on_spotfleet.git

Initializing provider plugins...
- Checking for available provider plugins on https://releases.hashicorp.com...
- Downloading plugin for provider "aws" (1.0.0)...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.aws: version = "~> 1.0"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```
```
hugo:lint/ $ tflint
ERROR: module `github.com/wata727/tf_aws_ecs_on_spotfleet` not found. Did you run `terraform get`?
```

Following, some info as you requested:

```
hugo:lint/ $ terraform --version
Terraform v0.10.7
```
```
hugo:lint/ $ tflint --version
TFLint version 0.4.3
```
```
hugo:lint/ $ ls -l .terraform/modules
total 4
drwxrwxr-x 4 hugo hugo 4096 Oct 10 14:08 89ec5104783f13faad3cf3d42b337fe3
```

Hope this help in something.

**Comment by wata727** [comment]
@ryanking @hposca Thanks for the great help!
Surprisingly, module directory path seems to be changed in the new Terraform version again.

```
# v0.10.7
drwxr-xr-x 4 wata727 wata727 4096 Oct 11 14:03 89ec5104783f13faad3cf3d42b337fe3

# v0.10.6
drwxr-xr-x 4 wata727 wata727 4096 Oct 11 14:10 1ebd7b7b95afeafe7e756740774adf56

# v0.10.5
drwxr-xr-x 4 wata727 wata727 4096 Oct 11 14:13 0c7a0e460217539236f583f3e6a187f2
```

I will fix it soon as soon as possible.

**Comment by khosrow** [comment]
The bug still seems to be there with the combination below:

```
> terraform --version
Terraform v0.11.0

> tflint --version
TFLint version 0.5.2
```

Here's my configuration:

**variables.tf**
```
variable "aws_access_key" {}
variable "aws_secret_key" {}
variable "aws_region" {}
```

**main.tf**
```
provider "aws" {
    access_key = "${var.aws_access_key}"
    secret_key = "${var.aws_secret_key}"
    region     = "${var.aws_region}"
}

module "vpc" {
    source = "terraform-aws-modules/vpc/aws"

    name = "test-vpc"
    cidr = "10.1.0.0/16"

    azs              = ["us-west-2a"]
    public_subnets   = ["10.1.1.0/24"]

    create_database_subnet_group = false
    enable_nat_gateway = true
    enable_dns_support = true
}
```

And the error I get:
```
> terraform get
- module.vpc
  Found version 1.3.0 of terraform-aws-modules/vpc/aws on registry.terraform.io
  Getting source "terraform-aws-modules/vpc/aws"

> tflint
ERROR: module `terraform-aws-modules/vpc/aws` not found. Did you run `terraform get`?
```

**Comment by acim** [comment]
Yes, the bug is still present as tflint doesn't find my local module even though terraform correctly creates resources from the module.

Terraform v0.11.0
+ provider.aws v1.2.0
+ provider.bitbucket v0.1.0
+ provider.template v1.0.0

TFLint version 0.5.2

`tflint
ERROR: module ./bitbucket not found. Did you run terraform get?

terraform get
- module.dns
- module.net
- module.bitbucket
- module.net.ssh
- module.net.http
- module.net.ssh-out-bitbucket
- module.net.vpc`

**Comment by dkatona** [comment]
Still doesn't work for me with Terraform 0.11.1 and TFlint 0.5.3. When I run tflint, the error is not produced anymore, but it doesn't detect any errors/typos that it should. When I run tflint -d, it produces a bunch of errors like this:
`DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_instance_not_specified_iam_profile`
DEBUG:/detector.go:116 [INFO] detect module `development_vpc_setup`
DEBUG:/detector.go:119 [ERROR] ERROR: module `github.com/WanderaOrg/terraform-aws-vpc` not found. Did you run `terraform get`?
Awesome! Your code is following the best practices :)`

I ran terraform get before and the module is present in .terraform/modules.

**Comment by wata727** [comment]
@dkatona Thanks for the feedback.
Is `github.com/WanderaOrg/terraform-aws-vpc`defined in the nested module?  Looking at the log, the module appears to be called in `development_vpc_setup` module. Unfortunately, TFLint doesn't support nested module detection.

**Comment by juldou** [comment]
It doesn't work for me when running it in Docker (seems it didn't see downloaded .terraform file from `terraform get` command). If I run it with installed tflint with brew, it's ok.

**Comment by pagameba** [comment]
Also did not work for me with the docker image, same issue (modules not found).

**Comment by ssanbeg** [comment]
The bug is still present.

terraform get
- module.vpc
- ...

tflint
ERROR: module `terraform-aws-modules/vpc/aws` not found. Did you run `terraform get`?

module "vpc" {
  source  = "terraform-aws-modules/vpc/aws"
  version = "1.60.0"
  ...

terraform -version
Terraform v0.11.13
+ provider.aws v2.7.0
+ provider.null v2.1.1

tflint --version
TFLint version 0.7.5
